test/FastTests/Corax/CompoundFieldsOnIndex.cs

```diff
@@ -0,0 +1,155 @@
+            Assert.Equal(3, users.Count);
+            Assert.Equal(2009, users[0].Birthday.Year);
+            Assert.Equal(2014, users[1].Birthday.Year);
+            Assert.Equal(2014, users[2].Birthday.Year);
+            
+            Assert.Equal("A", users[1].Name);
+            Assert.Equal("B", users[2].Name);
+        }
+        
+        using (var s = store.OpenSession())
+        {
+            var users = s.Query<User, Users_Idx>()
+                .Where(x => x.Location == "Hadera")
+                .OrderBy(x => x.Name)
+                .ToList();
+            Assert.Equal(2, users.Count);
+            Assert.Equal("Corax", users[0].Name);
+            Assert.Equal("Lucene", users[1].Name);
+        }
+    }
+
+    [RavenFact(RavenTestCategory.Querying)]
+    public async Task CanOptimizeToSkipSorting()
+    {
+        await TestQueryBuilder<MultiTermMatch>(s => s.Advanced.AsyncDocumentQuery<User, Users_Idx>()
+            .WhereEquals(x => x.Location, "Hadera")
+            .OrderBy(x => x.Name)
+            .GetIndexQuery()
+        );
+        await TestQueryBuilder<MultiTermMatch>(s => s.Advanced.AsyncDocumentQuery<User, Users_Idx>()
+            .WhereEquals(x => x.Name, "Corax")
+            .OrderBy(x => x.Birthday)
+            .GetIndexQuery()
+        );
+    }
+
+    [RavenFact(RavenTestCategory.Querying)]
+    public async Task Will_NOT_OptimizeQueryIfThereIsNoMatchingCompoundIndex()
+    {
+        await TestQueryBuilder<SortingMatch>(s => s.Advanced.AsyncDocumentQuery<User, Users_Idx>()
+            .WhereEquals(x => x.Name, "Lucene")
+            .AndAlso()
+            .WhereEquals(x=>x.Location, "Hadera")
+            .OrderBy(x => x.Birthday)
+            .GetIndexQuery()
+        );
+        
+        await TestQueryBuilder<SortingMatch>(s => s.Advanced.AsyncDocumentQuery<User, Users_Idx>()
+            .WhereEquals(x => x.Name, "Lucene")
+            .OrderBy(x => x.Location)
+            .GetIndexQuery()
+        );
+        
+        await TestQueryBuilder<SortingMultiMatch>(s => s.Advanced.AsyncDocumentQuery<User, Users_Idx>()
+            .WhereEquals(x => x.Location, "Hadera")
+            .OrderBy(nameof(User.Name))
+            .OrderBy(nameof(User.Birthday))
+            .GetIndexQuery()
+        );
+    }
+
+    private Task TestQueryBuilder<TExpected>(Func<IAsyncDocumentSession, IndexQuery> query)
+    {
+        return StreamingOptimization_QueryBuilder.TestQueryBuilder<TExpected,Users_Idx>(this, false, query);
+    }
+
+
+    private record User(string Name, string Location, DateTime Birthday);
+
+    private class Users_Idx : AbstractIndexCreationTask<User>
+    {
+        public Users_Idx()
+        {
+            Map = users => 
+                from u in users 
+                select new { u.Name, u.Location, u.Birthday };
+
+            CompoundFields.Add(new[] { nameof(User.Name), nameof(User.Birthday) });
+            CompoundFields.Add(new[] { nameof(User.Location), nameof(User.Name) });
```

Comment:
I would rather not do that on the client side. Since this is runtime checked anyway, we will have the `CompoundField` method with two args, so that is already checked.
And a way to add more if needed. We'll absolutely like to have more fields allowed in the future.

Reply:
Done in https://github.com/ravendb/ravendb/pull/17176/commits/04d16c6516dd9974c952589f8d049b5907c90e0d